### PR TITLE
Getaddrinfo fail timeout

### PR DESCRIPTION
**Pull request by calid**
On my machine `getaddrinfo_fail` consistently times out:

<pre>
=============================================================
&#91;%  61|+ 125|-   0|T   0|S   0]: getaddrinfo_fail
`getaddrinfo_fail` failed: timeout
Output from process `getaddrinfo_fail`: (no output)
=============================================================
&#91;% 100|+ 203|-   1|T   0|S   0]: Done.
FAIL: test/run-tests
</pre>

Tracing through the call stack showed that `getaddrinfo` was taking just a bit longer than the 5 second timeout to give up on the fake hostname:

<pre>
$ test/run-tests getaddrinfo_fail getaddrinfo_fail
getaddrinfo_fail start:     1391183554.344
uv_getaddrinfo start:       1391183554.344
uv_getaddrinfo end:         1391183554.344
uv__getaddrinfo_work start: 1391183554.344
<b>getaddrinfo start:          1391183554.344</b>
uv__io_poll start:          1391183554.344
uv__epoll_wait start:       1391183554.344
<b>getaddrinfo end:            1391183559.457</b>
uv__getaddrinfo_work end:   1391183559.457
uv__epoll_wait end:         1391183559.457
uv__async_io start:         1391183559.457
uv__async_event start:      1391183559.457
uv__work_done start:        1391183559.457
uv__getaddrinfo_done start: 1391183559.457
uv__getaddrinfo_done end:   1391183559.457
getaddrinfo_fail_cb start:  1391183559.457
getaddrinfo_fail_cb end:    1391183559.457
uv__work_done end:          1391183559.457
uv__async_event end:        1391183559.457
getaddrinfo_fail end:       1391183559.457
</pre>

In this patch I've upped the timeout to 10000 to compensate.  I've also modified the test harness to support setting timeouts on a per-test basis (e.g. _only_ getaddrinfo_fail has a 10000 timeout, the other tests remain at the default).

I worked my way through the contribution guidelines, so hopefully everything is in order.  Let me know if there's anything I missed.

Thank you!

### CODE REVIEW

**Comment by calid** [comment]
I did in fact sign the CLA, so not sure if Jenkins just needs some time to update?

File: test/test-list.h [hunk 1]
```diff
@@ -248,7 +248,7 @@
 
 
 TASK_LIST_START
-  TEST_OUTPUT_ENTRY  (platform_output)
+  TEST_ENTRY_CUSTOM (platform_output,0,1,5000)
```
Comment (indutny) [inline test/test-list.h]: Very minor issue, but I'll be glad if you'll add spaces after commas.

**Comment by indutny** [comment]
This looks very clean and neat! Thank you very much for contributing it, could you please sign a CLA so we could proceed to landing it?

**Comment by indutny** [comment]
Yeah, after checking it - I found your name in it. So please fix one minor nit and we are ready to go.

**Comment by calid** [comment]
@indutny no problem, fix pushed

**Comment by calid** [comment]
@indutny looking at the Jenkins output looks like I missed updating the benchmark code (I only ran the tests locally). Let me try to get that squared away as well.

**Comment by calid** [comment]
@indutny I think we're squared away, let me know if there are any further problems.

**Comment by indutny** [comment]
You forgot to fix `TASK_LIST_END` and `HELPER_ENTRY`, but nvm, I'll do it myself.

**Comment by calid** [comment]
ah sorry, thought I was safe when both tests and benchmarks passed... guess not :)

**Comment by indutny** [comment]
Thank you, landed in 756087e0171b33da512e1939290e24e1cc93d9a9 and 6c525df76d1e732c197f926f931cc804925392ac

**Pull request closed without merging**